editor/animation_track_editor.cpp

```diff
@@ -556,7 +556,8 @@ class AnimationTrackKeyEdit : public Object {
 			} break;
 			case Animation::TYPE_METHOD: {
 				p_list->push_back(PropertyInfo(Variant::STRING, "name"));
-				p_list->push_back(PropertyInfo(Variant::INT, "arg_count", PROPERTY_HINT_RANGE, "0,5,1"));
+				static_assert(VARIANT_ARG_MAX == 8, "PROPERTY_HINT_RANGE needs to be updated if VARIANT_ARG_MAX != 8");
+				p_list->push_back(PropertyInfo(Variant::INT, "arg_count", PROPERTY_HINT_RANGE, "0,8,1"));
```

Comment:
We discussed this in a PR review meeting and we're not sure it's worth changing the whole `VARIANT_ARG_MAX` just for this - the limit on animation code doesn't seem related to `VARIANT_ARG_MAX` and could likely be lifted to allow any number of arguments, no?

Reply:
In this code the user is modifying a key within a method-type track of an `AnimationPlayer`, right? The number of arguments for a method-type track in an `AnimationPlayer` is bound by the maximum number of arguments the `AnimationPlayer` itself supports when playing this track. `VARIANT_ARG_MAX` is related. I fail to see how it could not.  Keep an eye on `Animation::TYPE_METHOD`.

When the AnimationPlayer attempts to play the method-type track, it will run the following code (only pay attention to the first line, the last line, and `params` near the bottom):
https://github.com/godotengine/godot/blob/b197de6f5f1e61a9cecf0a29aadea0fb024b1edf/scene/animation/animation_player.cpp#L527-L548

The very last line of that code block has a direct reference to `VARIANT_ARG_MAX`. That check is there because further below `AnimationPlayer` reconstructs the method call extracted from the track:

https://github.com/godotengine/godot/blob/b197de6f5f1e61a9cecf0a29aadea0fb024b1edf/scene/animation/animation_player.cpp#L556-L573
The `params` array holds the number of arguments previously saved on the track through the `AnimationTrackEditor`. Everytime `VARIANT_ARG_MAX` is updated, that last code block needs to be updated to support the maximum number of arguments.